gitops/yaml/resources.go

```diff
@@ -75,3 +75,20 @@ func MarshalOutput(out io.Writer, output interface{}) error {
 	}
 	return nil
 }
+
+// The following is implemented by redhat-appstudio/application-service
+
+// UnMarshalItemFromFile unmarshals item from file
+func UnMarshalItemFromFile(fs afero.Fs, filename string, item interface{}) error {
+	content, err := afero.ReadFile(fs, filename)
+	if err != nil {
+		return fmt.Errorf("failed to read from file %s: %v", filename, err)
+	}
+
+	err = yaml.Unmarshal(content, item)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal data: %v", err)
+	}
```

Comment:
we can get test coverage % for this block by just trying to mimic a test case from go pkg:
<img width="1142" alt="Screen Shot 2022-07-18 at 4 59 37 PM" src="https://user-images.githubusercontent.com/31771087/179616510-2b7fcb71-4514-4465-b5a5-774e880cc93b.png">

Reply:
added test coverage for the unmarshal data and also marshal data failing case